In [1]:
# General load and data prep 
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import matplotlib.ticker as mticker
import seaborn as sns

sns.set_style('whitegrid')
plt.rcParams['figure.dpi'] = 100
pd.set_option('display.max_columns', 50)
pd.set_option('display.float_format', lambda x: f'{x:,.2f}')


df = pd.read_csv('WA_Fn-UseC_-HR-Employee-Attrition.csv')
df['Attrition_Flag'] = (df['Attrition'] == 'Yes').astype(int)
attrition_rate = df['Attrition_Flag'].mean()

# Tenure groups
df['TenureGroup'] = pd.cut(
    df['YearsAtCompany'],
    bins=[-1, 2, 5, 10, 100],
    labels=['0-2 yrs', '3-5 yrs', '6-10 yrs', '11+ yrs']
)

# Income quartiles (Q1 = lowest income, Q4 = highest)
df['IncomeQuartile'] = pd.qcut(df['MonthlyIncome'], 4, labels=['Q1', 'Q2', 'Q3', 'Q4'])


In [2]:
import pandas as pd

df = pd.read_csv('WA_Fn-UseC_-HR-Employee-Attrition.csv')
df['Attrition_Flag'] = (df['Attrition'] == 'Yes').astype(int)
df['IncomeQuartile'] = pd.qcut(df['MonthlyIncome'], 4, labels=['Q1', 'Q2', 'Q3', 'Q4'])

baseline_rate = df['Attrition_Flag'].mean()

# Target segment: OverTime=Yes AND JobSatisfaction=1 AND lowest income quartile (Q1)
mask = (
    (df['OverTime'] == 'Yes') &
    (df['JobSatisfaction'] == 1) &
    (df['IncomeQuartile'] == 'Q1')
)
segment = df[mask]

n = len(segment)
rate = segment['Attrition_Flag'].mean()

print(f'Segment size (n)        : {n}')
print(f'Segment attrition rate  : {rate:.1%}')
print(f'Company baseline rate   : {baseline_rate:.1%}')
print(f'Risk multiplier (lift)  : {rate / baseline_rate:.1f}x')

Segment size (n)        : 23
Segment attrition rate  : 65.2%
Company baseline rate   : 16.1%
Risk multiplier (lift)  : 4.0x


	                           n	 Attrition Rate	     vs. Baseline
Lowest income quartile (Q1)	   23	   65.2%	              4.0x
Below-median income	           41	   46.3%	              2.9x